File: frontend/src/typings/record/consultationRecord.ts
export interface IConsultationRecordResponse {
  id: number;
  sessionId: string;
  uid: string;
  title: string;
  urgency: 'green' | 'yellow' | 'red' | string;
  summary: string;
  messageCount: number;
  startedAt?: string | null;
  endedAt?: string | null;
  generatedAt?: string | null;
}

export interface IConsultationRecordPageResponse {
  records: IConsultationRecordResponse[];
  total: number;
  page: number;
  pageSize: number;
  hasMore: boolean;
}

File: services/cardiology-cloud/cardiology-record/src/main/java/com/sherwinzeng/cardiology/cardiologyrecord/entity/ConsultationRecord.java
package com.sherwinzeng.cardiology.cardiologyrecord.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("consultation_record")
public class ConsultationRecord {

    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    private String sessionId;
    private String uid;
    private String title;
    private String urgency;
    private String summary;
    private Integer messageCount;
    private LocalDateTime startedAt;
    private LocalDateTime endedAt;
    private LocalDateTime generatedAt;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
